---
title: "Workshop"
author: "Awtum"
date: "July 12, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


load libraries and datasets
#source("http://bioconductor.org/biocLite.R")
#biocLite("Biobase")
#biocLite("genefilter")
#devtools::install_github("genomicsclass/GSE5859")
```{r library load}
library(Biobase)
library(rafalib)
library(genefilter)
library(GSE5859)
data(GSE5859)
```
## R markdown report 1

### gene expression levels  
```{r}
geneExpression = exprs(e)

```
print date info
```{r}
sampleInfo = pData(e)
head(sampleInfo$date)
```
Output: print a table of year vs ethnic groups
```{r}
year = factor( format(sampleInfo$date,"%y") )
tab = table(year,sampleInfo$ethnicity)
print(tab)
```
###Differential expression analysis
remove control genes
```{r}
out <- grep("AFFX",rownames(geneExpression))
```

Use a t-test to perform differential expression analysis 
between CEU and ASN
```{r}
eth <- sampleInfo$ethnicity
ind<- which(eth%in%c("CEU","ASN"))
res1 <- rowttests(geneExpression[-out,ind],droplevels(eth[ind]))
```
Use a t-test to perform differential expression analysis between 2002 vs 2003 within CEU
```{r}
ind <- which(year%in%c("02","03") & eth=="CEU")
res2 <- rowttests(geneExpression[-out,ind],droplevels(year[ind]))
```
output: plot histograms for p-values 
```{r}
hist(res1$p, main="p-value of CEU vs ASN")
hist(res2$p, main="p-value of 2002 vs 2003 for CEU")
```

